protocol/messenger_communities.go

```diff
@@ -1869,3 +1869,149 @@ func (m *Messenger) RequestExtractDiscordChannelsAndCategories(filesToImport []s
 		m.config.messengerSignalsHandler.DiscordCategoriesAndChannelsExtracted(response.DiscordCategories, response.DiscordChannels, int64(response.DiscordOldestMessageTimestamp))
 	}()
 }
+
+func (m *Messenger) ImportDiscordCommunity(request *requests.ImportDiscordCommunity) (*MessengerResponse, error) {
+
+	discordCategories, exportedData, _, err := m.ExtractDiscordDataFromImportFiles(request.FilesToImport)
+	if err != nil {
+		return nil, err
+	}
+
+	createCommunityRequest := request.ToCreateCommunityRequest()
+	response, err := m.CreateCommunity(createCommunityRequest, false)
+	if err != nil {
+		return nil, err
+	}
+	response.CommunityChanges = []*communities.CommunityChanges{}
+
+	// We can safely assume that there's at least one community
+	// in the `MessengerResponse` because we've just created it
+	discordCommunity := response.Communities()[0]
+	addedCategoriesIds := make(map[string]string, 0)
+
+	for _, category := range discordCategories {
+		createCommunityCategoryRequest := &requests.CreateCommunityCategory{
+			CommunityID:  discordCommunity.ID(),
+			CategoryName: category.Name,
+			ChatIDs:      make([]string, 0),
+		}
+
+		r, err := m.CreateCommunityCategory(createCommunityCategoryRequest)
+		if err != nil {
+			return nil, err
+		}
+		response.AddCommunities(r.Communities())
+
+		// We can safely assume that there's at least one `CommunityChanges`
+		// in the `MessengerResponse` because we've just added a category
+		addedCategoryChange := r.CommunityChanges[0]
+		for _, addedCategory := range addedCategoryChange.CategoriesAdded {
+			addedCategoriesIds[category.ID] = addedCategory.CategoryId
+		}
+	}
+
+	messagesToSave := make([]*common.Message, 0)
+
+	for _, exportedChannel := range exportedData {
+
+		communityChat := &protobuf.CommunityChat{
+			Permissions: &protobuf.CommunityPermissions{
+				Access: protobuf.CommunityPermissions_NO_MEMBERSHIP,
+			},
+			Identity: &protobuf.ChatIdentity{
+				DisplayName: exportedChannel.Channel.Name,
+				Emoji:       "😎",
```

Comment:
@John-44 We need to decide what the default emoji of an imported discord channel is going to be